astropy/utils/misc.py

```diff
@@ -922,12 +922,40 @@ def size(self):
     def isscalar(self):
         return self.shape == ()
 
+    def __len__(self):
+        if self.isscalar:
+            raise TypeError("Scalar {0!r} object has no len()"
+                            .format(self.__class__.__name__))
+        return self.shape[0]
+
+    def __bool__(self):  # Python 3
+        """Any instance should evaluate to True, except when it is empty."""
+        return self.size > 0
+
+    def __nonzero__(self):  # Python 2
```

Comment:
Just a minor thing, you could also use:

```
if six.PY2:
    __nonzero__ = __bool__
    del __bool__  # optional
```

makes it easier to find and remove if python2 support is dropped.